distribution/std-lib/Base/src/Data/Vector.enso

```diff
@@ -210,6 +230,25 @@ type Vector
     map function =
         here.new this.length i-> function (this.at i)
 
+    ## Applies a function to each element of the vector, returning the vector
+       that contains all results concatenated.
+
+       > Example
+         In the following example, we replace each number `n` with itself
+         repeated `n` times:
+            [0, 1, 2] . flat_map (n -> Vector.fill n n)
+         The result of running the code above is:
+            [1, 2, 2]
+    flat_map : (Any -> Vector) -> Vector
+    flat_map function =
+        mapped = this.map function
+        length = mapped.map .length . fold 0 (+)
```

Comment:
Do it with a single fold. We don't have stream fusion yet, so you'd materialize an array of ints just for this intermediate value.